contest/standings.py

```diff
@@ -44,11 +44,29 @@ def test_func(self):
 
   def get_queryset(self):
     if self.virtual_progress is not None:
-      return get_contest_rank(self.contest, self.virtual_progress)
-    return get_contest_rank(self.contest)
+      rank_list = get_contest_rank(self.contest, self.virtual_progress)
+    else:
+      rank_list = get_contest_rank(self.contest)
+
+    if 'q' in self.request.GET:
+      contest_participants = {user.user_id: user for user in
+                              ContestParticipant.objects.filter(contest=self.contest).select_related('user').all()}
+      self.search_text = self.request.GET['q']
+      ret_list = []
+      for item in rank_list:
+        user = contest_participants[item['user']]
+        if not user.comment.find(self.search_text) == -1 \
```

Comment:
The filter can be directly done in query, which is more efficient.

Reply:
OK